cpp/perspective/src/cpp/sparse_tree.cpp

```diff
@@ -1127,19 +1127,13 @@ t_stree::update_agg_table(t_uindex nidx, t_agg_update_info& info,
                 new_value.set(reduce_from_gstate<
                     std::function<t_tscalar(std::vector<t_tscalar>&)>>(gstate,
                     expression_master_table, spec.get_dependencies()[0].name(),
-                    pkeys, [](std::vector<t_tscalar>& values) {
+                    pkeys, [&](std::vector<t_tscalar>& values) {
                         if (values.size() == 0) {
                             return t_tscalar();
                         } else if (values.size() == 1) {
                             return values[0];
                         } else {
-                            std::vector<t_tscalar>::iterator middle
-                                = values.begin() + (values.size() / 2);
-
-                            std::nth_element(
-                                values.begin(), middle, values.end());
-
-                            return *middle;
+                            return get_aggregate_median(values);
```

Comment:
Just a suggestion - if we're going to factor out this logic into a function called `get_aggregate_median(values)`, maybe we should also move the `0` and `1` cases into this definition to 1) make it complete and 2) move the entire closure body to a single method call.

Reply:
Done!